gradio/components.py

```diff
@@ -5330,6 +5331,94 @@ def as_example(self, input_data: str | None) -> str:
         return postprocessed if postprocessed else ""
 
 
+@document("change")
+class Code(Changeable, IOComponent, SimpleSerializable):
+    """
+    Creates a Code editor for entering, editing or viewing code.
+    Preprocessing: passes a {str} of code.
+    Postprocessing: expects the function to return a {str} filepath or a {str} of code.
+    """
+
+    languages = [
+        "python",
+        "markdown",
+        "json",
+        "html",
+        "css",
+        "javascript",
+        "typescript",
+        "yaml",
+        "dockerfile",
+        "shell",
+        "r",
+    ]
+
+    def __init__(
+        self,
+        value: str | None = None,
+        language: str | None = None,
+        *,
+        label: str | None = None,
+        show_label: bool = True,
+        visible: bool = True,
+        elem_id: str | None = None,
+        **kwargs,
+    ):
+        """
+        Parameters:
+            value: Default value. If callable, the function will be called whenever the app loads to set the initial value of the component.
+            language: The language to display the code as. Supported languages listed on `gr.Code.languages`.
+            label: component name in interface.
+            show_label: if True, will display label.
+            visible: If False, component will be hidden.
+            elem_id: An optional string that is assigned as the id of this component in the HTML DOM. Can be used for targeting CSS styles.
+        """
+        self.language = language
+        IOComponent.__init__(
+            self,
+            label=label,
+            show_label=show_label,
+            visible=visible,
+            elem_id=elem_id,
+            value=value,
+            **kwargs,
+        )
+
+    def get_config(self):
+        return {
+            "value": self.value,
+            "language": self.language,
+            **IOComponent.get_config(self),
+        }
+
+    def postprocess(self, y):
+        if y is not None and os.path.exists(y):
```

Comment:
I'd suggest using 
```suggestion
        if y is not None and os.path.isfile(y):
```
to eliminate directories